### PR TITLE
(maint) Update file source docs

### PR DESCRIPTION
**Pull request by joshcooper**
Document explicit `checksum_value` with `source`, `X-Checksum-*` headers,
userinfo component and HTTPS servers with `ssl_trust_store` setting.

### CODE REVIEW

File: lib/puppet/type/file/source.rb [hunk 1]
```diff
@@ -44,11 +44,26 @@ module Puppet
...
+      defaults to using the `Last-Modified` header. Puppet will update the local
+      file if the header is newer than the modified time (mtime) of the local
+      file.
+
+      _HTTP_ URIs can include a user information component so that Puppet can
+      retrieve file metadata and content from HTTP servers that require HTTP Basic
+      authentication. For example `https://<user>:<pass>@<server>:<port>/path/to/file.`
+
+      When connecting to _HTTPS_ servers, Puppet trusts CA certificates in the
+      puppet-agent certificate bundle and the Puppet CA. Puppet can be configured
```
Comment (clairecadman) [inline lib/puppet/type/file/source.rb:64]: > 

> 
```suggestion
      puppet-agent certificate bundle and the Puppet CA. You can configure Puppet to
```

File: lib/puppet/type/file/source.rb [hunk 2]
```diff
@@ -44,11 +44,26 @@ module Puppet
...
       ownership or permission details.
 
-      The `http` source uses the server `Content-MD5` header as a checksum to
-      determine if the remote file has changed. If the server response does not
-      include that header, Puppet defaults to using the `Last-Modified` header.
-      Puppet will update the local file if the header is newer than the modified
-      time (mtime) of the local file.
+      Puppet determines if file content is synchronized by computing a checksum
+      for the local file and comparing it against the `checksum_value`
+      parameter. If the `checksum_value` parameter is not specified, then for
```
Comment (clairecadman) [inline lib/puppet/type/file/source.rb:49]:
```suggestion
      parameter. If the `checksum_value` parameter is not specified for `puppet` and `file` sources, Puppet will 
```

File: lib/puppet/type/file/source.rb [hunk 3]
```diff
@@ -44,11 +44,26 @@ module Puppet
...
+      `puppet` and `file` sources, Puppet will compute a checksum based on its
+      `Puppet[:digest_algorithm]`. For `http(s)` sources, Puppet will use the
```
Comment (clairecadman) [inline lib/puppet/type/file/source.rb:51]:
```suggestion
      `Puppet[:digest_algorithm]`. For `http(s)` sources, Puppet uses the
```

File: lib/puppet/type/file/source.rb [hunk 4]
```diff
@@ -44,11 +44,26 @@ module Puppet
...
+      first HTTP header it recognizes out of the following list:
+      `X-Checksum-Sha256`, `X-Checksum-Sha1`, `X-Checksum-Md5` or `Content-MD5`.
+      If the server response does not include one of those headers, then Puppet
```
Comment (clairecadman) [inline lib/puppet/type/file/source.rb:54]:
```suggestion
      If the server response does not include one of these headers, Puppet
```

File: lib/puppet/type/file/source.rb [hunk 5]
```diff
@@ -44,11 +44,26 @@ module Puppet
...
+      defaults to using the `Last-Modified` header. Puppet will update the local
```
Comment (clairecadman) [inline lib/puppet/type/file/source.rb:55]:
```suggestion
      defaults to using the `Last-Modified` header. Puppet updates the local
```

File: lib/puppet/type/file/source.rb [hunk 6]
```diff
@@ -44,11 +44,26 @@ module Puppet
...
+      file if the header is newer than the modified time (mtime) of the local
+      file.
+
+      _HTTP_ URIs can include a user information component so that Puppet can
+      retrieve file metadata and content from HTTP servers that require HTTP Basic
```
Comment (clairecadman) [inline lib/puppet/type/file/source.rb:60]:
```suggestion
      retrieve file metadata and content from HTTP servers that require HTTP basic
```

**Comment by joshcooper** [comment]
Thanks for the feedback @clairecadman! I'll update this PR later today

**Comment by joshcooper** [comment]
I think I got all of your suggestions accept for `HTTP Basic Authentication` as it's the name of a specific type of authentication: https://en.wikipedia.org/wiki/Basic_access_authentication

**Pull request merged**